### PR TITLE
add kill command

### PR DESCRIPTION
**Pull request by JeromeVi**
I added the code to do `docker-machine kill myMachine` using node-docker-machine

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -82,6 +82,20 @@ class Machine {
...
+        done(new Error(`Docker host "${name}" does not exist`))
+      } else if (ALREADY_STOPPED.test(err)) {
+        done()
+      } else {
+        done(err)
+      }
+    })
+  }    
+    
+
```
Comment (vweevers) [inline index.js]: Please remove this double newline.

File: index.js [hunk 2]
```diff
@@ -223,4 +237,4 @@ function merge(node, data) {
 
 function isObject(obj) {
   return typeof obj === 'object' && obj !== null
-}
+}
```
Comment (vweevers) [inline index.js:238]: And revert this change (restore the newline).

**Review by vweevers — CHANGES_REQUESTED** [review]

**Comment by vweevers** [comment]
In addition to the two nitpicks regarding newlines, please:

1. Add unit tests. Luckily you can simply copy the [tests for `stop()`](https://github.com/vweevers/node-docker-machine/blob/089a93600ae3609c28ecf5ac3ec95d35d8ebbe53/test/index.js#L128-L161) and replace every instance of the word "stop" with "kill".
2. Add `'kill'` to the list of methods on [index.js#L205](https://github.com/vweevers/node-docker-machine/blob/089a93600ae3609c28ecf5ac3ec95d35d8ebbe53/index.js#L205)

**Comment by JeromeVi** [comment]
I don't understand how it works, if you want I copy the code here and you do the change

File: test/index.js [hunk 3]
```diff
@@ -160,6 +160,41 @@ test('stop', function (t) {
...
+  const s3 = spy({ error: new Error('\nhost does not exist abc') })
+  const s4 = spy({ error: new Error('\nalready stopped abc') })
+  const s5 = spy({ error: new Error('other error') })
+
+  Machine.kill('beep', (err) => {
+    t.ifError(err, 'no kill error')
+    t.same(s1.args, ['kill', 'beep'])
+  })
+
+  new Machine().stop(err => {
```
Comment (vweevers) [inline test/index.js]: `stop` should be `kill`

**Comment by JeromeVi** [comment]
Is it ok now ?

**Comment by JeromeVi** [comment]
Could you tell me if something is wrong ? Because I would appreciate to use the docker-machine kill method 👍

**Comment by JeromeVi** [comment]
Any news about this PR ?

File: test/index.js [hunk 4]
```diff
@@ -160,6 +160,41 @@ test('stop', function (t) {
...
+
+  const s1 = spy({})
+  const s2 = spy({})
+  const s3 = spy({ error: new Error('\nhost does not exist abc') })
+  const s4 = spy({ error: new Error('\nalready stopped abc') })
+  const s5 = spy({ error: new Error('other error') })
+
+  Machine.kill('beep', (err) => {
+    t.ifError(err, 'no stop error')
+    t.same(s1.args, ['stop', 'beep'])
```
Comment (vweevers) [inline test/index.js]: Another leftover "stop" here.

**Comment by vweevers** [comment]
Before you request feedback again, please make sure the unit tests pass. And squash the commits into one commit, with a descriptive message like "add kill command".

**Comment by JeromeVi** [comment]
Ok I'm boring to spam you and I understand, I feel myself boring too to do that but the checks failed and I don't know why because if I take a look in the details it's a method who fails but at no moment I have modified this line. 
```
:\projects\node-docker-machine\test\index.js:51
    t.is(s2.opts.cwd, '.', 'cwd')
                ^
TypeError: Cannot read property 'cwd' of undefined
    at C:\projects\node-docker-machine\test\index.js:51:17
    at nextTickCallbackWith2Args (node.js:458:9)
    at process._tickCallback (node.js:372:17)
npm ERR! Test failed.  See above for more details.
```

The only things I did were:
1) adding the kill test in /test/index.js
```
test('kill', function (t) {
  t.plan(10)

  const s1 = spy({})
  const s2 = spy({})
  const s3 = spy({ error: new Error('\nhost does not exist abc') })
  const s4 = spy({ error: new Error('\nalready stopped abc') })
  const s5 = spy({ error: new Error('other error') })

  Machine.kill('beep', (err) => {
    t.ifError(err, 'no kill error')
    t.same(s1.args, ['kill', 'beep'])
  })

  new Machine().kill(err => {
    t.ifError(err, 'no kill error')
    t.same(s2.args, ['kill', 'default'])
  })

  Machine.kill('boop', (err) => {
    t.is(err.message, 'Docker host "boop" does not exist', 'non existent error')
    t.same(s3.args, ['kill', 'boop'])
  })

  new Machine().kill(err => {
    t.ifError(err, 'no kill error if already stopped')
    t.same(s4.args, ['kill', 'default'])
  })

  Machine.kill('four', (err) => {
    t.is(err.message, 'other error', 'passthrough other error')
    t.same(s5.args, ['kill', 'four'])
  })
})
```

2) adding the kill method in index.js 
```
  static kill(name, done) {
    Machine.command(['kill', name], (err) => {
      if (HOST_NON_EXISTENT.test(err)) {
        done(new Error(`Docker host "${name}" does not exist`))
      } else if (ALREADY_STOPPED.test(err)) {
        done()
      } else {
        done(err)
      }
    })
  }   
```

+ in the list:
```
;['status', 'isRunning', 'start', 'stop', 'kill', 'env', 'ssh', 'inspect'].forEach(method => {
  Machine.prototype[method] = function () {
    const args = Array.from(arguments)
    args.unshift(this.name)
    Machine[method].apply(null, args)
  }
})
```

**Comment by vweevers** [comment]
@JeromeVi okay, I still hope you can get unstuck, otherwise, I will take care of this on Sunday.

**Comment by vweevers** [comment]
Closed in favor of #10.

**Pull request closed without merging**